src/Protocol/Diaspora.php

```diff
@@ -46,6 +47,35 @@
  */
 class Diaspora
 {
+	/**
+	 * Mark the relay contact of the given contact for archival
+	 * This is called whenever there is a communication issue with the server.
+	 * It avoids sending stuff to servers who don't exist anymore.
+	 * The relay contact is a technical contact entry that exists once per server.
+	 *
+	 * @param array $contact of the relay contact
+	 */
+	public static function markRelayForArchival($contact)
+	{
+		if (!empty($contact['contact-type']) && ($contact['contact-type'] == Contact::TYPE_RELAY)) {
+			// This is already the relay contact, we don't need to fetch it
+			$relay_contact = $contact;
+		} elseif (empty($contact['baseurl'])) {
+			if (!empty($contact['batch'])) {
+				$relay_contact = DBA::selectFirst('contact', [], ['batch' => $contact['batch'], 'contact-type' => Contact::TYPE_RELAY, 'uid' => 0]);
```

Comment:
Do we really need all fields? It seems like we used the relay contact just to mark it archival. As far as I know it's better to use explicit fields during a select instead of `*` but I don't exactly know it for MySQL

Reply:
We don't need all files. But I wanted to maintain the same behaviour at all places where we call the ```markForArchival``` function.

Reply:
Ok, I just thought we can add here a performance enhancement to not load the whole line every time. But sounds reasonable to leave the behaviour as is (and we're in the RC phase, so better stick to working patterns :) )